Emulator/BF.File.Emulator/Bf/BfBuilder.cs

```diff
@@ -0,0 +1,175 @@
+﻿using FileEmulationFramework.Lib.IO;
+using FileEmulationFramework.Lib.Utilities;
+using FileEmulationFramework.Lib.Memory;
+using AtlusScriptLibrary.FlowScriptLanguage.Compiler;
+using FlowFormatVersion = AtlusScriptLibrary.FlowScriptLanguage.FormatVersion;
+using AtlusScriptLibrary.Common.Libraries;
+using System.Text;
+using Microsoft.Win32.SafeHandles;
+using AtlusScriptLibrary.FlowScriptLanguage;
+using Newtonsoft.Json;
+
+// Aliasing for readability, since our assembly name has priority over 'File'
+using Fiel = System.IO.File;
+
+
+namespace BF.File.Emulator.Bf
+{
+    public class BfBuilder
+    {
+
+        private readonly List<string> _flowFiles = new List<string>();
+        private readonly List<string> _msgFiles = new List<string>();
+        private readonly List<FlowScriptModuleFunction> _libraryFuncs = new List<FlowScriptModuleFunction>();
+        private readonly List<FlowScriptModuleEnum> _libraryEnums = new List<FlowScriptModuleEnum>();
+        private readonly HashSet<string> _addedOverrides = new();
+        
+        private FlowFormatVersion? _flowFormat = null;
+        private Library? _library = null;
+        private Encoding? _encoding = null;
+
+        public BfBuilder(FlowFormatVersion? flowFormat, Library? library, Encoding? encoding)
```

Comment:
Hint: You can probably clean this up by adding logger as constructor parameter; then not passing it in every method.

Reply:
Done in 5874ac9